rust/private/rust.bzl

```diff
@@ -546,30 +548,27 @@ def _rust_test_impl(ctx):
 
 def _rust_library_group_impl(ctx):
     dep_variant_infos = []
-    dep_variant_transitive_infos = []
     runfiles = []
 
     for dep in ctx.attr.deps:
-        if rust_common.crate_info in dep:
-            dep_variant_infos.append(rust_common.dep_variant_info(
-                crate_info = dep[rust_common.crate_info] if rust_common.crate_info in dep else None,
-                dep_info = dep[rust_common.dep_info] if rust_common.crate_info in dep else None,
+        if CrateInfo in dep:
+            dep_variant_infos.append(DepVariantInfo(
+                crate_info = dep[CrateInfo] if CrateInfo in dep else None,
+                dep_info = dep[DepInfo] if DepInfo in dep else None,
                 build_info = dep[BuildInfo] if BuildInfo in dep else None,
                 cc_info = dep[CcInfo] if CcInfo in dep else None,
                 crate_group_info = None,
             ))
-        elif rust_common.crate_group_info in dep:
-            dep_variant_transitive_infos.append(dep[rust_common.crate_group_info].dep_variant_infos)
+        elif CrateGroupInfo in dep:
+            dep_variant_infos.extend(dep[CrateGroupInfo].dep_variant_infos)
```

Comment:
we could dedupe here if we think folks have nested `rust_library_groups` with duplicated deps